src/Library/Library.js

```diff
@@ -19,22 +21,51 @@ import { build } from "../../troll/src/builder.js";
 
 export default function Library({ application }) {
   const objects = build(resource);
-  const { window, picture_illustration, search_entry } = objects;
+  const {
+    window,
+    picture_illustration,
+    search_entry,
+    dropdown_category,
+    dropdown_language,
+  } = objects;
   window.application = application;
   picture_illustration.set_resource(illustration);
+  makeDropdownFlat(dropdown_category);
+  makeDropdownFlat(dropdown_language);
 
   if (__DEV__) {
     window.add_css_class("devel");
   }
 
   let last_triggered;
+  let current_category = 0;
+  let current_language = 0;
 
   window.connect("close-request", quitOnLastWindowClose);
 
   const demos = getDemos();
   const widgets_map = new Map();
   const category_map = new Map();
+  const language_model = new Gtk.StringList();
+  const category_model = new Gtk.StringList();
+  language_model.append("All Languages");
+  category_model.append("All Categories");
+  const language_check = ["All Languages"];
+  const category_check = ["All Categories"];
```

Comment:
We have not discussed localization of these strings, but I agree that the category strings are not ideal. How do we localize them?

Reply:
I think the simplest thing to do is keep a static map, since there's a fixed amount of strings, like:

```js
let language_labels = {
  "javascript": _("JavaScript"),
  // etc
};
```

There seem to be strings for all languages in `Extensions.blp`, so the same strings can be reused without increasing translatables. It seems like the preference is to keep translations in `.blp` files, but I don't see an elegant way to keep them there with the `Library.js` parts being dynamic.

Reply:
Okay, so I'll do that for both categories and languages. translatable category strings are present in the `Library.blp` except `_("Uncategorized");`
The linter does not like these changes 
```
  58:5   error  Replace `"javascript"` with `javascript`          prettier/prettier
  58:19  error  '_' is not defined                                no-undef
  59:5   error  Replace `"python"` with `python`                  prettier/prettier
  59:15  error  '_' is not defined                                no-undef
  60:5   error  Replace `"rust"` with `rust`                      prettier/prettier
```

Reply:
Adding this import should get the messages translated:

```js
import { gettext as _ } from "gettext";
```

I think the prettier style probably just wants the double-quotes removed from the dictionary keys, since they're not actually necessary.